packages/amplify_storage_s3/lib/amplify_storage_s3.dart

```diff
@@ -36,7 +36,7 @@ class AmplifyStorageS3 extends StoragePluginInterface {
 
   static set instance(StoragePluginInterface instance) {
     PlatformInterface.verifyToken(instance, _token);
-    _instance = instance;
+    _instance = instance as AmplifyStorageS3;
```

Comment:
Is there any reason the setter should accept type `StoragePluginInterface` instead of type `AmplifyStorageS3`? The instance should always be of type `AmplifyStorageS3` in this class, right?. If there is a scenario where that is not true, this cast is going to throw an exception. If it is true, I think we should just update the getter/setter.

```dart
  static AmplifyStorageS3 get instance => _instance;

  static set instance(AmplifyStorageS3 instance) {
    PlatformInterface.verifyToken(instance, _token);
    _instance = instance;
  }
```

Reply:
Thanks, same question.

Reply:
Yes you're right though I would say the instance should always be of interface type.  

We want to accept the interface type to allow for other implementations to be accepted.